src/transaction/flashback.h

```diff
@@ -55,6 +55,49 @@
     } \
   while (0)
 
+#define FLASHBACK_SET_MIN_LOG_PAGEID_TO_KEEP(lsa_p) \
+  do \
+    { \
+      if ((lsa_p) != NULL) \
+        { \
+          flashback_Min_log_pageid = (lsa_p)->pageid; \
+        } \
+    } \
+  while (0)
+
+#define FLASHBACK_UNSET_MIN_LOG_PAGEID_TO_KEEP() \
+  do \
+    { \
+      flashback_Min_log_pageid = NULL_LOG_PAGEID; \
+    } \
+  while (0)
+
+/* set threshold to 60 which is recommended intervals for removing archive logs */
+#define FLASHBACK_SET_THRESHOLD() \
+  do \
+    { \
+      flashback_Threshold_to_remove_archive = prm_get_integer_value (PRM_ID_REMOVE_LOG_ARCHIVES_INTERVAL); \
+      if (flashback_Threshold_to_remove_archive == 0) \
+        { \
+          flashback_Threshold_to_remove_archive = 60; \
+        } \
+    } \
+  while (0)
+
+#define FLASHBACK_SET_REQUEST_TIME() \
+  do \
+    { \
+      flashback_Last_request_time = time (NULL); \
+    } \
+  while (0)
+
+extern LOG_PAGEID flashback_Min_log_pageid;	// Minumun log pageid to keep archive log volume from being removed
+extern time_t flashback_Last_request_time;	// The time most recently requested by flashback
+extern int flashback_Threshold_to_remove_archive;	/* If the difference between the time at which the archive log is deleted
```

Comment:
In my opinion, it would be better to make these static global variables and provide interfaces to set them instead of macros. They don't have to be exposed to other modules. You already define getters. Could you tell me the reason for making setters as macros and variables accessible from outside?

Reply:
Oh.. yes you are right. Thank you. I will change it to static global variables, and provide setter function.
It was my mistake to set these variable as global variables that can be accessed from outside.